crates/bevy_ecs/src/event/mod.rs

```diff
@@ -1,39 +1,322 @@
 //! Event handling types.
-mod base;
-mod collections;
-mod event_cursor;
-mod iterators;
-mod mut_iterators;
-mod mutator;
-mod reader;
-mod registry;
-mod update;
-mod writer;
-
-pub(crate) use base::EventInstance;
-pub use base::{BufferedEvent, EntityEvent, Event, EventId, EventKey};
+mod buffered_event;
+mod trigger;
+
 pub use bevy_ecs_macros::{BufferedEvent, EntityEvent, Event};
-#[expect(deprecated, reason = "`SendBatchIds` was renamed to `WriteBatchIds`.")]
-pub use collections::{Events, SendBatchIds, WriteBatchIds};
-pub use event_cursor::EventCursor;
-#[cfg(feature = "multi_threaded")]
-pub use iterators::EventParIter;
-pub use iterators::{EventIterator, EventIteratorWithId};
-#[cfg(feature = "multi_threaded")]
-pub use mut_iterators::EventMutParIter;
-pub use mut_iterators::{EventMutIterator, EventMutIteratorWithId};
-pub use mutator::EventMutator;
-pub use reader::EventReader;
-pub use registry::{EventRegistry, ShouldUpdateEvents};
-#[expect(
-    deprecated,
-    reason = "`EventUpdates` was renamed to `EventUpdateSystems`."
-)]
-pub use update::{
-    event_update_condition, event_update_system, signal_event_update_system, EventUpdateSystems,
-    EventUpdates,
+pub use buffered_event::*;
+pub use trigger::*;
+
+use crate::{
+    component::{Component, ComponentId},
+    entity::Entity,
+    world::World,
 };
-pub use writer::EventWriter;
+use core::marker::PhantomData;
+
+/// An [`Event`] is something that "happens" at given point.
```

Comment:
```suggestion
/// An [`Event`] is something that "happens" at a given point.
```

Maybe "point in time"? Either way, this is a typo.